src/main/java/duke/Ui.java

```diff
@@ -0,0 +1,165 @@
+package duke;
+
+import java.io.IOException;
+import java.util.Scanner;
+
+import javafx.application.Application;
+import javafx.scene.Scene;
+import javafx.scene.control.Button;
+import javafx.scene.control.Label;
+import javafx.scene.control.ScrollPane;
+import javafx.scene.control.TextField;
+import javafx.scene.layout.AnchorPane;
+import javafx.scene.layout.Border;
+import javafx.scene.layout.BorderStroke;
+import javafx.scene.layout.BorderStrokeStyle;
+import javafx.scene.layout.BorderWidths;
+import javafx.scene.layout.CornerRadii;
+import javafx.scene.layout.HBox;
+import javafx.scene.layout.Region;
+import javafx.scene.layout.VBox;
+import javafx.scene.paint.Paint;
+import javafx.stage.Stage;
+
+/**
+ * Manages the exchange of user input and program output.
+ */
+public class Ui extends Application {
+    public static final String LOGO = "D U K E";
+    protected Scanner scanner;
+    protected TaskList tasks;
+    private Stage stage;
+    private ScrollPane scrollPane;
+    private VBox dialogContainer;
+    private TextField userInput;
+    private Button sendButton;
+    private Scene scene;
+
+    /**
+     * Initialises the input scanner and task list.
+     */
+    public void initialise() {
+        scanner = new Scanner(System.in);
+
+        // Initialise task list
+        tasks = new TaskList();
+        try {
+            tasks.load();
+        } catch (IOException ex) {
+            System.out.println("  Unable to load tasks.");
+        }
+    }
+
+    /**
+     * Creates the GUI for user to communicate with Duke.
+     *
+     * @param stage The stage for the GUI.
+     */
+    @Override
+    public void start(Stage stage) {
+        this.stage = stage;
+        initialise();
+
+        // Create nodes for GUI
+        scrollPane = new ScrollPane();
+        dialogContainer = new VBox();
+        scrollPane.setContent(dialogContainer);
+
+        userInput = new TextField();
+        sendButton = new Button("Send");
+
+        AnchorPane mainLayout = new AnchorPane();
+        mainLayout.getChildren().addAll(scrollPane, userInput, sendButton);
+
+        scene = new Scene(mainLayout);
+        stage.setScene(scene);
+        stage.show();
+
+        // Arrange nodes for GUI
+        stage.setTitle("Duke");
+        stage.setResizable(false);
+        stage.setMinHeight(600.0);
+        stage.setMinWidth(400.0);
```

Comment:
Perhaps extracting these magic numbers out as named constants will be easier to understand?

Reply:
Thanks, will do so.